system/datatypes/peer_reviewed_article.php

```diff
@@ -139,6 +139,25 @@ public function process(HyphaRequest $request) {
 		return '404';
 	}
 
+	public function getSortDateTime() {
+		$article = $this->xml->find(self::FIELD_NAME_ARTICLE);
+		$value = $article->getAttribute(self::FIELD_NAME_PUBLISHED_AT);
+		if (!$value)
+			$value = $article->getAttribute(self::FIELD_NAME_UPDATED_AT);
+		$timestamp = ltrim($value, "t");
+		return new DateTime("@" . $timestamp);
+	}
+
+	public function renderExcerpt($container) {
+		$doc = $container->document();
+		$doc->createElement('h2')->appendTo($container)->setText($this->getTitle());
+		$article = $this->xml->find(self::FIELD_NAME_ARTICLE);
+		$this->appendAuthorAndTimestamp($container, $article);
+		$excerpt_div = $doc->createElement('div')->appendTo($container);
```

Comment:
Same comment as before. I would suggest first building the element and then add it to the DOM

Reply:
Fixed.